src/Term/EntityTerms.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Wikibase\DataModel\Term;
+
+use Comparable;
+use InvalidArgumentException;
+
+/**
+ * Imutable value object.
+ *
+ * @since 3.1.0
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class EntityTerms implements Comparable {
+
+	/**
+	 * @var string
+	 */
+	private $languageCode;
+
+	/**
+	 * @var Term
+	 */
+	private $label;
+
+	/**
+	 * @var Term
+	 */
+	private $description;
+
+	/**
+	 * @var AliasGroup
+	 */
+	private $aliases;
+
+	/**
+	 * @param string $languageCode
+	 * @param Term|null $label
+	 * @param Term|null $description
+	 * @param AliasGroup|null $aliases
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	public function __construct(
+		$languageCode,
+		Term $label = null,
+		Term $description = null,
+		AliasGroup $aliases = null
+	) {
+		if ( !is_string( $languageCode ) || $languageCode === '' ) {
+			throw new InvalidArgumentException( '$languageCode must be a non-empty string' );
+		}
+
+		$this->languageCode = $languageCode;
+		$this->label = $label ?: new Term( $languageCode, '' );
```

Comment:
I understand that some kind of `hasLabelForLanguage` must be possible. But this is unrelated to the current state of this proposal.